System/Posix/Fcntl.hsc

```diff
@@ -101,3 +111,68 @@ foreign import capi safe "fcntl.h posix_fallocate"
 fileAllocate _ _ _ = ioError (ioeSetLocation unsupportedOperation
                               "fileAllocate")
 #endif
+
+-- -----------------------------------------------------------------------------
+-- File caching
+
+-- | Performs the @fcntl(2)@ operation on a file-desciptor to get the cache mode.
+--
+-- If the cache mode is 'False', then cache effects for file system reads and
+-- writes are minimised or otherwise eliminated. If the cache mode is 'True',
+-- then cache effects occur like normal.
+--
+-- On Linux, FreeBSD, and NetBSD this checks whether the @O_DIRECT@ file flag is
```

Comment:
Is OpenBSD different from other *BSD systems with regards to this?

Reply:
There is no POSIX standard for direct IO, so probably not all BSD variants support it, but at least FreeBSD and NetBSD explicitly do